src/runner/system_menu_helper.h

```diff
@@ -0,0 +1,54 @@
+#pragma once
+
+#pragma once
+
+#include <interface/powertoy_system_menu.h>
+#include <windows.h>
+#include <string>
+#include <vector>
+#include <unordered_map>
+
+class PowertoyModuleIface;
+
+class SystemMenuHelper : public PowertoySystemMenuIface
+{
+public:
+  // PowertoySystemMenuIface
+  virtual void SetConfiguration(PowertoyModuleIface* module, const wchar_t* config) override;
+  virtual void RegisterAction  (PowertoyModuleIface* module, HWND window, const wchar_t* name) override;
+
+  bool Customize(PowertoyModuleIface* module, HWND window);
+  void Reset    (PowertoyModuleIface* module);
+
+  bool HasCustomConfig(PowertoyModuleIface* module);
+
+  PowertoyModuleIface* ModuleFromItemId(const int& id);
+  const std::wstring   ItemNameFromItemId(const int& id);
+private:
+
+  /* Data parsed from JSON configuration suplied by module itself. */
+  struct ItemInfo {
+    std::wstring name{};
+    bool         enable{ false };
+    bool         check{ false };
+  };
+
+  bool AddItem(PowertoyModuleIface* module, HWND window, const std::wstring& name, const bool enable);
+  bool AddSeparator(PowertoyModuleIface* module, HWND window);
+  void ParseConfiguration(const std::wstring& config, std::vector<ItemInfo>& out);
+
+  /* Store processed windows to avoid handling it multiple times. */
+  std::unordered_map<HWND, std::vector<PowertoyModuleIface*>>            ProcessedWindows{};
```

Comment:
`ProcessedWindows` keeps track of all modules processed for a window, so the name should be `ProcessedModules` in fact it is used to check if for a specific windows we already processed a specific module.